src/fetchers/gist-fetcher.js

```diff
@@ -0,0 +1,84 @@
+// @ts-check
+
+import axios from "axios";
+import { parseHTML } from "linkedom";
+
+/**
+ * @typedef {import('./types').GistData} GistData Gist data.
+ */
+
+/**
+ * Fetch GitHub gist information by given username and ID.
+ *
+ * @param {string} username Github username.
+ * @param {string} id Github gist ID.
+ * @returns {Promise<GistData>} Gist data.
+ */
+const fetchGist = async (username, id) => {
+  const response = (
+    await axios({
+      method: "get",
+      url: `https://api.github.com/gists/${id}`,
+      headers: {
+        "Content-Type": "application/json",
+        Accept: "application/vnd.github.cloak-preview",
+      },
+    })
+  ).data;
+  const { starsCount, forksCount } = await fetchGistStargazers(username, id);
+
+  return {
+    name: response.files[Object.keys(response.files)[0]].filename,
+    nameWithOwner: `${username}/${
+      response.files[Object.keys(response.files)[0]].filename
+    }`,
+    description: response.description,
+    language: response.files[Object.keys(response.files)[0]].language,
```

Comment:
I understand why you use the language of the first file but a gist can have multiple files. Maybe we can use the language of the file that has the biggest size?

Reply:
I wrote little function for calculating primary language by files size. 

https://github.com/anuraghazra/github-readme-stats/pull/3064/files#diff-85d91b5e80c17d5aa6499442fb72b2f5cc1a263e176ccba2187f106163584603R85-R103

Tell me what do you think

Reply:
Amazing! I checked your code, and I think it is very well-coded 👍🏻. I think for consistency it would be nice:

- If we add a snapshot test (see https://github.com/anuraghazra/github-readme-stats/blob/master/tests/fetchTopLanguages.test.js).

Apart from that I think this pull request can be merged 🚀!